### DISCUSSION TITLE
Forward-like but for list of pairs of units

### DISCUSSION THREAD

**Discussion by sergeysova** [sergeysova/patronum#151] [Ideas]
## Now

```ts
forward({
  from: navigation.historyPush,
  to: app.navigation.historyPush,
});
forward({
  from: navigation.historyPushSearch,
  to: app.navigation.historyPushSearch,
});
forward({
  from: navigation.historyReplace,
  to: app.navigation.historyReplace,
});
```

## Proposal

```ts
forwardPairs([
  [navigation.historyPush, app.navigation.historyPush],
  [navigation.historyPushSearch, app.navigation.historyPushSearch],
  [navigation.historyReplace, app.navigation.historyReplace],
])
```

**Comment by AlexandrHoroshih** [comment]
I like the idea, but this name sounds weird to me.
Maybe something like `forwardMap` would fit better?

**Comment by risenxxx** [comment]
I think that `forwardMany` is a good option - it's clear and short.

`forwardMap` has some collisions with the functions like `useStoreMap`, where the word `map` is used for describing "mapping". Though this function has something like "mapping", I don't think this is the same as in the other functions.

**Comment by AlexandrHoroshih** [comment]
Also, maybe some additional enchanced api would make sense?

Like this:

```ts
forwardPairs({
  base, // Record<string, Unit>
  target, // Record<string, Unit>
  rule, // (basePropName: string): string => targetPropName
})

// and then just
forwardPairs({
  base: navigation,
  target: app.navigation,
  rule: (name) => name, // makes more sense as a default option with api like that
})
```

**Reply by risenxxx** [reply]
Looks like the edge-case =)

Yeah, it's good for @sergeysova example, but imagine the usage without "scopes" with units.

**Reply by AlexandrHoroshih** [reply]
I was thinking that this operator should support both options - if there are "scopes" and it is possible to have some "rule" to map them to each other - then this api would help
For all other cases there would be the first api with explicit relations
